intellij-plugin/src/main/kotlin/com/apollographql/ijplugin/inspection/ApolloMissingGraphQLDefinitionImportInspection.kt

```diff
@@ -0,0 +1,208 @@
+      val knownDefinitionNames = definitions.filterIsInstance<GQLNamed>().map { it.name }
+      val arguments = directiveElement.arguments?.argumentList.orEmpty()
+      for (argument in arguments) {
+        val argumentDefinition = directiveDefinition.arguments.firstOrNull { it.name == argument.name } ?: continue
+        val argumentTypeToImport = argumentDefinition.type.rawType().name.takeIf { it in knownDefinitionNames } ?: continue
+        if (!isImported(directiveElement, argumentTypeToImport, definitionsUrl)) {
+          val typeKind = getTypeKind(argumentTypeToImport)
+          holder.registerProblem(
+              argument,
+              ApolloBundle.message(message, typeKind, argumentTypeToImport),
+              highlightType,
+              ImportDefinitionQuickFix(typeKind = typeKind, elementName = argumentTypeToImport, definitions = definitions, definitionsUrl = definitionsUrl),
+          )
+        }
+      }
+    }
+  }
+}
+
+private fun getTypeKind(typeName: String): String {
+  val typeDefinition = NULLABILITY_DEFINITIONS.firstOrNull { it is GQLNamed && it.name == typeName } ?: return "unknown"
+  return ApolloBundle.message(
+      when (typeDefinition) {
+        is GQLDirectiveDefinition -> "inspection.missingGraphQLDefinitionImport.reportText.directive"
+        is GQLEnumTypeDefinition -> "inspection.missingGraphQLDefinitionImport.reportText.enum"
+        is GQLInputObjectTypeDefinition -> "inspection.missingGraphQLDefinitionImport.reportText.input"
+        is GQLScalarTypeDefinition -> "inspection.missingGraphQLDefinitionImport.reportText.scalar"
+        else -> return "unknown"
+      }
+  )
+}
+
+private class ImportDefinitionQuickFix(
+    val typeKind: String,
+    val elementName: String,
+    private val definitions: List<GQLDefinition>,
+    private val definitionsUrl: String,
+) : LocalQuickFix {
+  override fun getName() = ApolloBundle.message("inspection.missingGraphQLDefinitionImport.quickFix", typeKind, "'$elementName'")
+  override fun getFamilyName() = name
+
+  override fun availableInBatchMode() = false
+  override fun generatePreview(project: Project, previewDescriptor: ProblemDescriptor): IntentionPreviewInfo = IntentionPreviewInfo.EMPTY
+
+  override fun applyFix(project: Project, descriptor: ProblemDescriptor) {
+    if (!IntentionPreviewUtils.isIntentionPreviewActive()) project.telemetryService.logEvent(TelemetryEvent.ApolloIjMissingGraphQLDefinitionImportQuickFix())
+
+    val element = descriptor.psiElement.parentOfType<GraphQLDirective>(withSelf = true)!!
+    val schemaFiles = element.schemaFiles()
+    val linkDirective = schemaFiles.flatMap { it.linkDirectives(definitionsUrl) }.firstOrNull()
+
+    // Special case: also add a @catch directive to the schema if we're importing @catch
+    val catchDirectiveSchemaExtension = if (element.name == CATCH && schemaFiles.flatMap { it.schemaCatchDirectives() }.isEmpty()) {
+      createCatchDirectiveSchemaExtension(project)
```

Comment:
This sounds fishy. Choosing your default `@catch(to:)` is an important choice, I'm afraid adding it automatically could be a bit confusing. Sounds like this needs to be a separate inspection: `Missing default @catch(to)`?

On the other hand the ergonomics of `@catch` aren't finalized yet so might not be worth spending too much time on this. 

I would still err on the side of not adding this automatically. The next build will make an error but 🤷

Reply:
True, we probably don't want to make a default choice. Maybe an build error is better for now, it will force users to make the choice.